code-challenges.js

```diff
@@ -26,3 +54,31 @@ var hand2 = [5, 5, 3, 3, 4]
 var hand3 = [5, 5, 5, 5, 4]
 // Expected output: false
 
+//Write a function that takes an array
+const isFullHouse = (argumentArray) => {
+    //Create an empty array to hold the ranks from the argument array
+    answerArray = []
+    //Map over the argument array to push a single instance of each rank to the answer array
+    argumentArray.map(value => answerArray.includes(value) ? "" : answerArray.push(value))
+    //If the answerArray does not have a length of 2 return false for the entire function
+    if (answerArray.length !== 2) {
+        return false
+    }
+    // //If not, map over the elements of the answerArray to execute the following expession twice: 
+    // //Filter through the argument array - if the elements match the element of the answer array at the map's current index exactly twice return true for that iteration of map
+    // //If either iteration of the map function returns true (i.e. if the array returned by map includes the value true), then return true for the entire function 
+    else if (answerArray.map((value, index) => argumentArray.filter( value => value === answerArray[index]).length === 2).includes(true)) {
+        return true
+    }
+    // //Else return false for the entire function
+    else {
+        return false
+    }
+    //Here is the above logic in ternary form:
+    // return answerArray.length !== 2 ? false :  answerArray.map((value, index) =>  argumentArray.filter( value1 => value1 === answerArray[0]).length === 2).includes(true) ? true : false
+}
```

Comment:
this code is super clean!

the pseudocode on all of these solutions was fantastic!